lib/task_helper.rb

```diff
@@ -0,0 +1,48 @@
+require 'json'
+
+class TaskHelper
+  class Error < RuntimeError
+    attr_reader :kind, :details, :issue_code
+
+    def initialize(msg, kind, details = nil)
+      super(msg)
+      @kind = kind
+      @issue_code = issue_code
+      @details = details || {}
+    end
+
+    def to_h
+      { 'kind' =>  kind,
+        'msg' => message,
+        'details' => details }
+    end
+  end
+
+  def task(params = {})
+    msg = 'The task author must implement the `task` method in the task'
+    error = TaskHelper::Error.new(msg, 'tasklib/not-implemented')
+    STDOUT.print error.to_h.to_json
+    raise error
+  end
+
+  def self.run
+    params = JSON.parse(STDIN.read)
+
+    # This method accepts a hash of parameters to run the task, then executes
+    # the task. Unhandled errors are caught and turned into an error result.
+    # @param [Hash] params A hash of params for the task
+    # @return [Hash] The result of the task
+    result = begin new.task(params)
+             rescue StandardError => e
+               error = TaskHelper::Error.new(e.message,
+                                             'tasklib/standard-error',
+                                             e.backtrace)
+               STDOUT.print error.to_h.to_json
+               raise error
+             end
+    msg = 'The `task()` function did not return a hash as expected.' \
```

Comment:
Michael agreed with you. I thought it might be opaque to new users what was happening if the output didn't look right and could be frustrating, but I like the idea of printing a string if it's not a hash.